ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/MLMemoryManager.java

```diff
@@ -0,0 +1,164 @@
+     * @param actionListener action listener to process the response
+     */
+    public void createConversation(String name, String applicationType, ActionListener<CreateConversationResponse> actionListener) {
+        try {
+            client.execute(CreateConversationAction.INSTANCE, new CreateConversationRequest(name, applicationType), actionListener);
+        } catch (RuntimeException exception) {
+            actionListener.onFailure(exception);
+        }
+    }
+
+    /**
+     * Adds an interaction to the conversation indicated, updating the conversational metadata
+     * @param conversationId the conversation to add the interaction to
+     * @param input the human input for the interaction
+     * @param promptTemplate the prompt template used for this interaction
+     * @param response the Gen AI response for this interaction
+     * @param origin the name of the GenAI agent in this interaction
+     * @param additionalInfo additional information used in constructing the LLM prompt
+     * @param parentIntId the parent interactionId of this interaction
+     * @param traceNum the trace number for a parent interaction
+     * @param actionListener gets the ID of the new interaction
+     */
+    public void createInteraction(
+        String conversationId,
+        String input,
+        String promptTemplate,
+        String response,
+        String origin,
+        Map<String, String> additionalInfo,
+        String parentIntId,
+        Integer traceNum,
+        ActionListener<CreateInteractionResponse> actionListener
+    ) {
+        Preconditions.checkNotNull(conversationId);
+        Preconditions.checkNotNull(input);
+        Preconditions.checkNotNull(response);
+        // additionalInfo cannot be null as flat object
+        additionalInfo = (additionalInfo == null) ? new HashMap<>() : additionalInfo;
+        try {
+            client
+                .execute(
+                    CreateInteractionAction.INSTANCE,
+                    new CreateInteractionRequest(
+                        conversationId,
+                        input,
+                        promptTemplate,
+                        response,
+                        origin,
+                        additionalInfo,
+                        parentIntId,
+                        traceNum
+                    ),
+                    actionListener
+                );
+        } catch (RuntimeException exception) {
+            actionListener.onFailure(exception);
+        }
+    }
+
+    /**
+     * Get the interactions associate with this conversation, sorted by recency
+     * @param conversationId the conversation whose interactions to get
+     * @param lastNInteraction Return how many interactions
+     * @param actionListener get all the final interactions that are not traces
+     */
+    public void getFinalInteractions(String conversationId, int lastNInteraction, ActionListener<List<Interaction>> actionListener) {
```

Comment:
From the method name,  this method is to get the final interaction, but the comments says "the interactions associate with this conversation".  Which does this method do ?

Reply:
By reading code of `innerGetInteractions` of InteractionsIndex.java,  the "final interaction" means the interaction rather than the trace data. Suggest explain more details in the comments

Reply:
added this explain in the comment title. Also the @param actionListener has that explain.